Lib/pdb.py

```diff
@@ -1672,6 +1703,22 @@ def _getsourcelines(self, obj):
         lineno = max(1, lineno)
         return lines, lineno
 
+    def _help_message_from_doc(self, doc):
+        lines = [line.strip() for line in doc.rstrip().splitlines()]
+        if not lines:
+            return "No help message found."
+        if "" in lines:
+            usage_end = lines.index("")
+        else:
+            usage_end = 1
+        usage = "\n".join([f"{' ' * len(self.prompt)}Usage: {lines[0].strip()}"] + \
+                [f"{' ' * len(self.prompt + 'Usage: ')}"
+                 f"{line.strip()}" for line in lines[1:usage_end]])
+
+        description = "\n".join(f"      {line.strip()}" for line in lines[usage_end:])
+
+        return f"{usage}\n{description}"
```

Comment:
I personally find something like this quite a bit more readable (I think it's equivalent). What do you think?

```suggestion
        formatted = []
        indent = " " * len(self.prompt)
        for i, line in enumerate(lines):
            if i == 0:
                prefix = "Usage: "
            elif i < usage_end:
                prefix = "       "
            else:
                prefix = ""
            formatted.append(indent + prefix + line)
        return "\n".join(formatted)
```

I also think your current implementation is double-stripping every line, and the `description` assumes a prompt length of 6.

Reply:
Yes, this is definitely easier to read.